synapse/cortex.py

```diff
@@ -298,16 +300,21 @@ def _convert_reqdict(reqdict):
         cron = self.cell.agenda.appts.get(iden)
         if cron is None:
             raise s_exc.NoSuchIden()
-        self._auth_check(cron.username)
+        self._trig_auth_check(cron.useriden)
         await self.cell.agenda.mod(iden, query)
 
     async def listCronJobs(self):
         '''
         Get information about all the cron jobs accessible to the current user
         '''
-        username = None if self.user is None else self.user.name
-        return [(iden, cron) for (iden, cron) in self.cell.agenda.list()
-                if username is None or self.user.admin or cron.get('user') == username]
+        crons = []
+        for iden, cron in self.cell.agenda.list():
+            if not (self.user.admin or cron['useriden'] == self.user.iden):
+                continue
+            cron['username'] = self.cell.auth.user(cron['useriden']).name
```

Comment:
same problem as noted above - what happens here if the user is gone from the hive?

Reply:
Fixed same way.